### PR TITLE
Add endpoint field in configuration page

### PR DESCRIPTION
**Pull request by FlorentinGarnier**
Add a list of endpoint in config page

### CODE REVIEW

File: src/Form/Type/PayzenGatewayConfigurationType.php [hunk 1]
```diff
@@ -36,6 +36,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'sylius.form.gateway_configuration.payzen.ctx_mode.choices.production' => Api::MODE_PRODUCTION,
                 ),
             ])
+            ->add('endpoint', ChoiceType::class, [
+                'label'       => 'sylius.form.gateway_configuration.payzen.endpoint.label',
+                'choices'  => array(
+                    'sylius.form.gateway_configuration.payzen.endpoint.choices.default' => null,
```
Comment (gplanchat) [inline src/Form/Type/PayzenGatewayConfigurationType.php:42]: Is the `sylius.form.gateway_configuration.payzen.endpoint.choices.default` option implicitly refers to https://secure.payzen.eu/vads-payment/ ?

File: src/Form/Type/PayzenGatewayConfigurationType.php [hunk 2]
```diff
@@ -36,6 +36,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                    'sylius.form.gateway_configuration.payzen.endpoint.choices.clic_and_pay' => Api::ENDPOINT_CLICANDPAY,
+                    'sylius.form.gateway_configuration.payzen.endpoint.choices.scellius' => Api::ENDPOINT_SCELLIUS,
+                    'sylius.form.gateway_configuration.payzen.endpoint.choices.systempay' => Api::ENDPOINT_SYSTEMPAY,
```
Comment (gplanchat) [inline src/Form/Type/PayzenGatewayConfigurationType.php:45]: In this list is missing SogeCommerce's URL: https://sogecommerce.societegenerale.eu/vads-payment/

File: src/Form/Type/PayzenGatewayConfigurationType.php (hunk 1 above)
Comment (FlorentinGarnier) [reply src/Form/Type/PayzenGatewayConfigurationType.php:42]: Yes it's the fall back of the API of ekyna/PayzenPayum

File: src/Form/Type/PayzenGatewayConfigurationType.php (hunk 2 above)
Comment (FlorentinGarnier) [reply src/Form/Type/PayzenGatewayConfigurationType.php:45]: Unfortunaly, SogeCommerce is not yet implemented in ekyna/PayzenPayum

File: src/Form/Type/PayzenGatewayConfigurationType.php [hunk 3]
```diff
@@ -36,6 +36,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                ),
```
Comment (Prometee) [inline src/Form/Type/PayzenGatewayConfigurationType.php]: @FlorentinGarnier can you add those missing translations into the `src/Resources/translations/messages.*.yaml`

**Review by Prometee — CHANGES_REQUESTED** [review]

File: src/Form/Type/PayzenGatewayConfigurationType.php (hunk 3 above)
Comment (FlorentinGarnier) [reply src/Form/Type/PayzenGatewayConfigurationType.php]: Sorry some files was not added in my last commit :)

File: src/Form/Type/PayzenGatewayConfigurationType.php (hunk 2 above)
Comment (Prometee) [reply src/Form/Type/PayzenGatewayConfigurationType.php:45]: Waiting for this PR to be merged and released : https://github.com/ekyna/PayumPayzen/pull/5

**Review by Prometee — APPROVED** [review]

File: src/Form/Type/PayzenGatewayConfigurationType.php (hunk 2 above)
Comment (Prometee) [reply src/Form/Type/PayzenGatewayConfigurationType.php:45]: I would say go to merge this since ekyna has made a new release

**Review by gplanchat — APPROVED** [review]

**Comment by Prometee** [comment]
About the build error, I didn't spent much time to investigate on it within my other Sylius plugin repositories, but it's something we don't have to care here, only Sylius 1.10 with Symfony 4.4 is failing.

**Comment by FlorentinGarnier** [comment]
@gplanchat Do you want resolve the CI issues ? Or wait an another PR ?

**Comment by Prometee** [comment]
No need to fix the build @FlorentinGarnier, Sylius 1.10 will be soon deprecated so we can only focus on Sylius > 1.11
But if you have an idea on how to resolve the error I'm open to discuss it 😁

**Comment by FlorentinGarnier** [comment]
> No need to fix the build @FlorentinGarnier, Sylius 1.10 will be soon deprecated so we can only focus on Sylius > 1.11 But if you have an idea on how to resolve the error I'm open to discuss it 😁

LGTM in this case ;)

**Comment by Prometee** [comment]
For the record, I found what was the issue with the missing service, it appears that `api_platform.error_listener` is relying on `exception_listener` service which is remove when you are not in a web context (eg: console here).

To fix it I alter a little bit the `tests/Application/Kernel.php` class by adding this :

```php
    protected function build(ContainerBuilder $container)
    {
        // Fix APIPlatform issue during console initialisation
        // Service "api_platform.error_listener": Parent definition "exception_listener" does not exist.
        $sfVersion = sprintf('%d.%d', BaseKernel::MAJOR_VERSION, BaseKernel::MINOR_VERSION);
        if ('4.4' !== $sfVersion) {
            return;
        }

        if (false === $container->hasDefinition('exception_listener')) {
            $container->setAlias('exception_listener', new Alias('console.error_listener'));
        }
    }
```

It will add an alias if the service is missing and only for Symfony 4.4.

**Pull request merged**